### PR TITLE
Regression tests for arithmetic proofs.

### PR DESCRIPTION
**Pull request by alex-ozdemir**
These tests are designed to test interesting cases of arithmetic proofs,
such as mixing integers and reals, equalities, and tightening bounds.

Right now, they have the --no-check-proofs flag set, which prevents them
from testing the proof machinery. However, once we check that machinery
into master, we'll remove that flag, thus enabling the full effect of
the tests.

### CODE REVIEW

File: test/regress/regress0/arith/arith-eq.smt2 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+; COMMAND-LINE: --no-check-models  --no-check-unsat-cores --no-check-proofs
```
Comment (yoni206) [inline test/regress/regress0/arith/arith-eq.smt2]: Is `--no-check-models` necessary?

Comment (alex-ozdemir) [reply test/regress/regress0/arith/arith-eq.smt2]: In haiku:
copy-paste strikes again

Comment (yoni206) [reply test/regress/regress0/arith/arith-eq.smt2]: :)
In that case, I also don't find the smtlib version that important either (I am assuming it is there for the same reason).

Comment (yoni206) [reply test/regress/regress0/arith/arith-eq.smt2]: BTW you have 6 syllables in the third line...

**Comment by alex-ozdemir** [comment]
> I ... would like to verify -- do all these tests terminate quickly

Yes. They take less than a second on my machine

**Comment by alex-ozdemir** [comment]
And sounds good. I'll remove those too.

**Pull request merged**